src/hybrid_reader.c

```diff
@@ -117,23 +122,55 @@ static void alternatingIterate(HybridIterator *hr, VecSimQueryResult_Iterator *v
   IndexResult_Free(cur_vec_res);
 }
 
-void prepareResults(HybridIterator *hr) {
-    if (hr->mode == STANDARD_KNN) {
+void computeDistances(HybridIterator *hr) {
+  float upper_bound = INFINITY;
+  RSIndexResult *cur_res = hr->base.current;
+  RSIndexResult *cur_child_res;  // This will use the memory of hr->child->current.
+  RSIndexResult *cur_vec_res = NewDistanceResult();
+
+  while (hr->child->Read(hr->child->ctx, &cur_child_res) != INDEXREAD_EOF) {
+    float dist = (float)VecSimIndex_GetDistanceFrom(hr->index, cur_child_res->docId, hr->query.vector);
+    // If this id is not in the vector index (since it was deleted), dist will return as NaN.
+    if (isnanf(dist)) {
+      continue;
+    }
+    if (heap_count(hr->topResults) < hr->query.k || dist < upper_bound) {
+      // Populate the vector result.
+      cur_vec_res->docId = cur_child_res->docId;
+      cur_vec_res->dist.distance = dist;
+      cur_vec_res->dist.scoreField = hr->scoreField;
+      insertResultToHeap(hr, cur_res, cur_child_res, cur_vec_res, &upper_bound);
+    }
+  }
+  IndexResult_Free(cur_vec_res);
+}
+
+static void prepareResults(HybridIterator *hr) {
+    if (hr->mode == VECSIM_STANDARD_KNN) {
       hr->list =
           VecSimIndex_TopKQuery(hr->index, hr->query.vector, hr->query.k, &(hr->runtimeParams), hr->query.order);
       hr->iter = VecSimQueryResult_List_GetIterator(hr->list);
     return;
   }
 
-  if (hr->mode == HYBRID_ADHOC_BF) {
+  if (hr->mode == VECSIM_HYBRID_ADHOC_BF) {
     // Go over child_it results, compute distances, sort and store results in topResults.
+    computeDistances(hr);
     return;
   }
   // Batches mode.
+  if (hr->child->NumEstimated(hr->child->ctx) == 0) {
+    return;
+  }
   VecSimBatchIterator *batch_it = VecSimBatchIterator_New(hr->index, hr->query.vector);
   float upper_bound = INFINITY;
   while (VecSimBatchIterator_HasNext(batch_it)) {
-    size_t batch_size = hr->query.k;  // todo: add heuristics here
+    hr->numIterations++;
```

Comment:
Where is it used?

Reply:
For monitoring only. Will move to `ft.profile`

Reply:
What does `numIterations` means here?
IMO it is better to increase every rewind. Maybe add additional `numRewind`?

Reply:
In the `fr.profile` I named it "Batches number" - it is the number of batches that were given from the vector index (which is also the number of rewinds that we did to the child iterator. I think that "Batches number" is a bit more informative than "numRewind". I will add that to the docs